src/python/pants/base/exception_sink.py

```diff
@@ -0,0 +1,88 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import datetime
+import logging
+import os
+import sys
+from builtins import object
+
+from pants.util.dirutil import is_writable_dir, safe_open
+
+
+logger = logging.getLogger(__name__)
+
+
+class ExceptionSink(object):
+  """A mutable singleton object representing where exceptions should be logged to."""
+
+  _destination = os.getcwd()
```

Comment:
Should probably explain why we do this ("blah blah increase the chances that if an exception occurs early in initialization we still record it somewhere").

Reply:
Absolutely agree, this is highly mysterious as is.